modules/dynamic-plug-in-api.adoc

```diff
@@ -0,0 +1,1467 @@
+
+[source,ts]
+----
+const HomePage: React.FC = (props) => {
+    const page = {
+      href: '/home',
+      name: 'Home',
+      component: () => <>Home</>
+    }
+    return <HorizontalNav match={props.match} pages={[page]} />
+}
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`resource` |The resource associated with this Navigation, an object of
+K8sResourceCommon type
+
+|`pages` |An array of page objects
+
+|`match` |match object provided by React Router
+|===
+
+[discrete]
+== `VirtualizedTable`
+
+A component for making virtualized tables.
+
+[source,ts]
+----
+const MachineList: React.FC<MachineListProps> = (props) => {
+  return (
+    <VirtualizedTable<MachineKind>
+     {...props}
+     aria-label='Machines'
+     columns={getMachineColumns}
+     Row={getMachineTableRow}
+    />
+  );
+}
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`data` |data for table
+|`loaded` |flag indicating data is loaded
+|`loadError` |error object if issue loading data
+|`columns` |column setup
+|`Row` |row setup
+|`unfilteredData` |original data without filter
+|`NoDataEmptyMsg` |(optional) no data empty message component
+|`EmptyMsg` |(optional) empty message component
+|`scrollNode` |(optional) function to handle scroll
+|`label` |(optional) label for table
+|`ariaLabel` |(optional) aria label
+|`gridBreakPoint` |sizing of how to break up grid for responsiveness
+|`onSelect` |(optional) function for handling select of table
+|`rowData` |(optional) data specific to row
+|===
+
+[discrete]
+== `TableData`
+
+Component for displaying table data within a table row.
+
+[source,ts]
+
+----
+const PodRow: React.FC<RowProps<K8sResourceCommon>> = ({ obj, activeColumnIDs }) => {
+  return (
+    <>
+      <TableData id={columns[0].id} activeColumnIDs={activeColumnIDs}>
+        <ResourceLink kind="Pod" name={obj.metadata.name} namespace={obj.metadata.namespace} />
+      </TableData>
+      <TableData id={columns[1].id} activeColumnIDs={activeColumnIDs}>
+        <ResourceLink kind="Namespace" name={obj.metadata.namespace} />
+      </TableData>
+    </>
+  );
+};
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`id` |unique id for table
+|`activeColumnIDs` |active columns
+|`className` |(optional) option class name for styling
+|===
+
+[discrete]
+== `useActiveColumns`
+
+A hook that provides a list of user-selected active TableColumns.
+
+[source,tsx]
+----
+  // See implementation for more details on TableColumn type
+  const [activeColumns, userSettingsLoaded] = useActiveColumns({
+    columns,
+    showNamespaceOverride: false,
+    columnManagementID,
+  });
+  return userSettingsAreLoaded ? <VirtualizedTable columns={activeColumns} {...otherProps} /> : null
+----
+
+[cols=",",options="header",]
+|===
+|Parameter Name |Description
+|`options` |Which are passed as a key-value map
+
+|`` |\{TableColumn[]} options.columns - An array of all available
```

Comment:
Is the formatting here correct?

Reply:
Hi, I'm working on the API documentation update in openshift/console based on your review ([jira link](https://issues.redhat.com/browse/OCPBUGS-5571)), I will be responding to some of your comments with my findings or questions. As the bad formatting you meant the part where the Parameter Name column contains the empty ``? I've seen that this is a common issue in our API documentation, is this a known issue? @spadgett @jhadvig